daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/PackedBinaryUnparserTraits.scala

```diff
@@ -45,7 +46,16 @@ abstract class PackedBinaryBaseUnparser(
   override def unparse(state: UState): Unit = {
     val nBits = getBitLength(state)
     val node = state.currentInfosetNode.asSimple
-    val value = node.dataValue.asInstanceOf[JNumber]
+
+    // If this is unparsing a binary calendar use node.dataValueAsString and create a JBigDecimal
+    //   so it can be cast to a JNumber below.
```

Comment:
It would be good to explain *why* we need to use dataValueAsString rather than dataValue (e.g. packed date time uses the convert combinator which sets the string value of the calendar--if we used dataValue we would get the Calendar value rather than that string. On the otherhand, with packed numbers, dataValue is already a number so we just want to use that, etc.).

Based on the code, it's clear that unparsing a packed binary calendar uses dataValueAsString, but it's not immediately clear why that's necessary.